src/components/common/Layout/components/Account/components/AccountModal/components/CopyCWaaSBackupData.tsx

```diff
@@ -0,0 +1,126 @@
+import { Icon, IconButton, Text, Tooltip, VStack } from "@chakra-ui/react"
+import { usePostHogContext } from "components/_app/PostHogProvider"
+import Button from "components/common/Button"
+import useDriveOAuth from "hooks/useDriveOAuth"
+import useSubmit from "hooks/useSubmit"
+import useToast from "hooks/useToast"
+import { Copy } from "phosphor-react"
+import { useState } from "react"
+import { getDriveFileAppProperties, listWalletsOnDrive } from "utils/googleDrive"
+
+const CopyCWaaSBackupData = () => {
+  const { captureEvent } = usePostHogContext()
+  const driveOAuth = useDriveOAuth()
+  const toast = useToast()
+  const [backup, setBackup] = useState<string>()
+
+  const tryToCopyBackup = (backupStr: string) => {
```

Comment:
The problem with that was, that it needs the value to be passed to the hook, not to the returned `onCopy`
But I think, what we could do is instead of triggering the toast inside the fetcher, we could just set it to the `backup` state variable, and throw the toast with a `useEffect`. That way we can just do `() => onCopy(backup)`, I'll give this a shot, thanks for pointing out!

Reply:
Just realised it has a `setValue` as well, so having the extra state might not even be needed

Reply:
Made these changes in https://github.com/guildxyz/guild.xyz/pull/1015/commits/a84256e9624a447d274e333588bba93999b5208a

I haven't mentioned it, but the reason for a bit of overcomplication with the toast is, that both the OAuth popup and the clipboard copy are actions that have to happen immediately (or very close) after an actual user click event. Lmk if you guys have some ideas on how else we could work this around